helpers/framed.js

```diff
@@ -0,0 +1,31 @@
+let framed;
+
+export async function isFramed() {
+	if (framed !== undefined) return framed;
+
+	if ((window.D2L && window.D2L.IsNotAnIframedApp) || window === window.parent) {
+		framed = false;
+		return framed;
+	}
+
+	const handleIsFramedResponse = (evt, resolve) => {
+		if (!evt || !evt.data || !evt.data.isFramed) return;
```

Comment:
Should the last check be `evt.data.isFramed === undefined`? I guess `isFramed` if it comes back at all is always `true` eh... yeah OK.

Reply:
Yeah, this is a weird one. The host will never respond with anything else. I can certainly make the check more specific just for correctness.